src/main/java/edu/ie3/datamodel/io/source/sql/SqlDataSource.java

```diff
@@ -177,11 +176,15 @@ protected Stream<Map<String, String>> buildStreamByTableName(String tableName) {
    * table name.
    */
   protected Stream<Map<String, String>> executeQuery(String query, AddParams addParams) {
-    try (PreparedStatement ps = connector.getConnection().prepareStatement(query)) {
+    try {
+      PreparedStatement ps = connector.getConnection().prepareStatement(query);
       addParams.addParams(ps);
 
-      ResultSet resultSet = ps.executeQuery();
-      return connector.extractFieldMaps(resultSet).stream();
+      Stream<Map<String, String>> stream = connector.toStream(ps, 1000);
+
+      // don't work with `try with resource`, but is necessary to get results
+      ps.closeOnCompletion();
```

Comment:
1. I think it'd make more sense to call this before the stream is created
2. Checking the JavaDoc of this method, I think this only works if the ResultSet is closed (which then causes the Statement to be closed).